configure.ac

```diff
@@ -430,6 +430,8 @@ MINGW=`echo $host_os | grep mingw32  2>/dev/null`
 AM_CONDITIONAL(MINGW, test ! -z "$MINGW" )
 if test ! -z "$MINGW"; then
 	WSOCKLIB="-lws2_32"
+	LDFLAGS="-no-undefined $LDFLAGS"
+	AC_DEFINE(HAVE_GETTIMEOFDAY)
```

Comment:
Can't say I'm happy either, but the main reasoning was better cross-platform compatibility (see comments in #153). To be honest, this PR may be one of those to prove them right... Does cmake work for you without any changes?